gammapy/maps/wcsnd.py

```diff
@@ -460,51 +460,150 @@ def downsample(self, factor, preserve_counts=True):
             WCS axis object
-        im : `~matplotlib.image.AxesImage`
-            Image object.
+        cbar : `~matplotlib.colorbar.Colorbar`
+            Colorbar object (if ``add_cbar=True`` was set, else ``None``)
+
         """
+        
         import matplotlib.pyplot as plt
-        import matplotlib.colors as colors
-
-        if ax is None:
+        from astropy.visualization import simple_norm
+        from scipy.ndimage import gaussian_filter
+        from scipy.signal import convolve
+        from scipy.stats import gmean
+        from astropy.convolution import Tophat2DKernel
+        from astropy.coordinates import Angle
+        
+        if fig is None:
             fig = plt.gcf()
-            ax = fig.add_subplot(111, projection=self.geom.wcs)
-
+        if ax is None:
+            ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
+        
         if idx is not None:
             idx = (idx,) if isinstance(idx, int) else idx
             slices = (slice(None), slice(None)) + idx
             data = self.data[slices[::-1]]
         else:
             data = self.data
 
-        kwargs.setdefault('interpolation', 'nearest')
+        if smooth is not None:
+            if isinstance(radius, Quantity):
+                val=Angle(np.abs(self.geom.wcs.wcs.cdelt),unit="deg")
+                radius=gmean(radius/val).value
+            if smooth == 'gauss':
+                width = radius / 2.
+                data = gaussian_filter(data, width, **kwargs)
+            if smooth == 'disk':
+                width = 2 * radius + 1
+                disk = Tophat2DKernel(width)
+                disk.normalize('integral')
+                data = convolve(data, disk.array, method='auto')
+
+        caxes = ax.imshow(data, **kwargs)
+
+        kwargs.setdefault('interpolation', 'None')
         kwargs.setdefault('origin', 'lower')
         kwargs.setdefault('norm', None)
+        kwargs.setdefault('cmap', 'afmhot')
+        norm = simple_norm(data[np.isfinite(data)], stretch)
+        kwargs.setdefault('norm', norm)
+
+
+        if add_cbar:
+            unit = self.geom.wcs.wcs.cunit or 'None'
+            label = self.geom.wcs.wcs.cname or None
+            #cbar = fig.colorbar(caxes, ax=ax, label='{} ({})'.format(label, unit))
+            cbar = fig.colorbar(caxes, ax=ax)
+        else:
+            cbar = None
+
+        try:
+            ax.coords['glon'].set_axislabel('Galactic Longitude')
+            ax.coords['glat'].set_axislabel('Galactic Latitude')
+        except KeyError:
+            ax.coords['ra'].set_axislabel('Right Ascension')
+            ax.coords['dec'].set_axislabel('Declination')
+        except AttributeError:
+            log.info("Can't set coordinate axes. No WCS information available.")
+
+        # without this the axis limits are changed when calling scatter
+        ax.autoscale(enable=False)
+        return fig, ax, cbar
+
+
+    def peek_at_pos(pos,axis,idx=None):
```

Comment:
Please remove this function, that is not implemented yet. You can add it any time in a second pull request.